src/stores/simulations.ts

```diff
@@ -16,18 +26,53 @@ export const useSimulationStore = defineStore('simulation', () => {
     }
   }
 
+  function setCurrentSubStep(subStep: number) {
+    if (subStep < 0) {
+      currentSubStep.value = 0
+    } else if (subStep > 2) {
+      currentSubStep.value = 2
+    } else {
+      currentSubStep.value = subStep
+    }
+  }
+
   function goToPreviousStep() {
```

Comment:
I think you can replace this function by something like this :

`function indexOfCurrentStepAndSubStep() {
    return availableSteps.findIndex((availableStep) => {
      return availableStep.step == currentStep.value && availableStep.subStep == currentSubStep.value
    })
  }

  function goToPreviousStep() {
    const currentIndex = indexOfCurrentStepAndSubStep()
    if(currentIndex-1 < 0) {
      return
    }
    setCurrentStep(availableSteps[currentIndex-1].step)
    setCurrentSubStep(availableSteps[currentIndex-1].subStep)
  }`